### PR TITLE
Documentation of Migrating from `ZManaged` to `Scope`

### PR DESCRIPTION
**Pull request by khajavi**

### CODE REVIEW

**Comment by adamgfraser** [comment]
There is some question here of how much we want to push people to the `managed` package versus just moving to scopes. I tend to think given how successful people have been in moving to scopes that we should push in that direction first. So an overall message like "Migrating to scopes is easy and you can do it like this ... if you have a lot of code that used `ZManaged` and you just don't want to deal with it right now you can do this ..."

**Review by jdegoes — APPROVED** [review]

**Comment by khajavi** [comment]
@adamgfraser 
Incorporated your feedback and added some more notes about the ZManaged solution and its drawbacks.

**Review by adamgfraser — CHANGES_REQUESTED** [review]
This looks really good. I think maybe we could provide a little more of a step by step path for migrating from `ZManaged` to `Scope`. Something like:

1. Replace all references to `ZManaged[R, E, A]` with `ZIO[R with Scope, E, A]`
2. Replace all references to `resource.use(f)` with `ZIO.scoped(resource.flatMap(f))`
3. Replace all `ZManaged` constructors with `ZIO.acquireRelease` or one of its variants
4. Replace all usages of `Layer(resource)` or `resource.toLayer` with `ZLayer.scoped(resource)`, all references  to `ZStream.managed(resource)` with `ZStream.scoped(resource)`, and so on for similar constructors
5. Delete all uses of `toManaged_`
6. Replace all uses of `toManaged(finalizer)` with `withFinalizer(finalizer)`

**Comment by khajavi** [comment]
@adamgfraser 
Thanks for your reviews.
Incorporated!

File: docs/howto/migrate/migration-guide.md [hunk 1]
```diff
@@ -1280,6 +1280,259 @@ Here is list of other deprecated methods:
...
+- def foo[R, E, A]: ZManaged[R, E, A] = ???
++ def foo[R, E, A]: ZIO[R with Scope, E, A] = ???
+```
+
+Example: 
+
+```diff
+trait ZPool[+Error, Item] {
+-  def make[R, E, A](get: => ZManaged[R, E, A], size: => Int): ZManaged[R, Nothing, ZPool[E, A]]
++  def make[R, E, A](get: => ZIO[R, E, A], size: => Int): ZIO[R with Scope, Nothing, ZPool[E, A]]
```
Comment (adamgfraser) [inline docs/howto/migrate/migration-guide.md:1367]: May want to use a different example here.

**Review by adamgfraser — APPROVED** [review]
Thanks for your work on this! This is going to make it a lot easier for people to upgrade to using scopes!

**Pull request merged**